front/src/modules/timesStops/hooks/useComputedMargins.tsx

```diff
@@ -0,0 +1,79 @@
+import type { SimulationResponse, TrainScheduleResult } from 'common/api/generatedEditoastApi';
+import type { OperationalPointWithTimeAndSpeed } from 'modules/trainschedule/components/DriverTrainScheduleV2/types';
+import { interpolateValue } from 'modules/trainschedule/components/DriverTrainScheduleV2/utils';
+
+import { marginRegExValidation } from '../consts';
+import { formatDigitsAndUnit } from '../utils';
+
+let marginUnit = '%';
+
+const useComputedMargins = (
+  trainSimulation: SimulationResponse | undefined,
+  opPoint: OperationalPointWithTimeAndSpeed,
+  nextOpPoint: OperationalPointWithTimeAndSpeed,
+  selectedTrainSchedule: TrainScheduleResult,
+  pathId: string
+) => {
+  if (trainSimulation?.status === 'success') {
+    const theoreticalMarginIndex = selectedTrainSchedule?.margins?.boundaries?.findIndex(
```

Comment:
You can test above if margins are defined, otherwise return the default values and also remove the ?
```suggestion
    const theoreticalMarginIndex = selectedTrainSchedule.margins.boundaries.findIndex(
```

Reply:
we have two reasons for failures, margins can be undefined, but we also can find no index. 
we can put that in a single if (theoreticalMarginIndex)

Reply:
```
  const theoreticalMarginIndex = selectedTrainSchedule?.margins?.boundaries?.findIndex(
    (id) => id === pathId
  );
  if (!theoreticalMarginIndex) {
    return ['', '', ''];
  }
```

Reply:
Yes I agree :) You can also remove the ? after selectedTrainSchedule

Reply:
done